Assignment-2-User-Management.c

```diff
@@ -0,0 +1,175 @@
+            fclose(fp);
+            printf("New user added \n");
+            break;
+        }
+        case 2: // to display users
+        {
+            FILE *fptr = fopen("user.txt", "r");
+
+            if (fptr == NULL)
+            {
+                printf("NO USER FOUND!\n");
+                break;
+            }
+
+            char str[100];
+            while (fgets(str, sizeof(str), fptr) != NULL)
+            {   
+                printf("%s", str);
+            }
+
+            fclose(fptr);
+            break;
+        }
+
+        case 3:  // to update users
+        {
+            int id, flag = 0;
+            printf("\nEnter the ID to update: ");
+            scanf("%d", &id);
+
+            FILE *file1 = fopen("user.txt", "r");  // for reading
+            FILE *writeFile = fopen("temp.txt", "w"); // for writing
+
+            if (file1 == NULL || writeFile == NULL)
+            {
+                printf("\n NO USER FOUND !\n");
+                break;
+            }
+
+            struct user newuser;
+            while (fscanf(file1, "%d,%[^,],%d\n", &newuser.id, newuser.name, &newuser.age) != EOF)
+            {
+                if (newuser.id == id)
+                {
+                    flag = 1;
+                    printf("Enter Updated Name: ");
+                    scanf(" %[^\n]", newuser.name);
+                    printf("Enter New Age: ");
+                    scanf("%d", &newuser.age);
+                }
+                fprintf(writeFile, "%d,%s,%d\n", newuser.id, newuser.name, newuser.age);
+            }
+
+            fclose(file1);
+            fclose(writeFile);
+
+            remove("user.txt");
+            rename("temp.txt", "user.txt");
+
+            if (flag)
+            {
+                printf("User updated successfully.\n");
+            }
+            else
+            {
+                printf(" USER NOT FOUND \n");
+            }
+            break;
+        }
+
+        case 4:
+        {
+            // to delete the users
+            int id, flag = 0;
+            printf("Enter the ID to delete: ");
+            scanf("%d", &id);
+
+            FILE *file = fopen("user.txt", "r"); 
+            FILE *tempFile = fopen("temp.txt", "w");
+
+            if (file == NULL || tempFile == NULL)
+            {
+                printf("PROBLEM IN OPENING THE FILE\n");
+                break;
+            }
+
+            struct user newuser;
+            while (fscanf(file, "%d,%[^,],%d\n", &newuser.id, newuser.name, &newuser.age) != EOF)
+            {
+                if (newuser.id == id)
+                {
+                    flag = 1;
+                    continue;
+                }
+                fprintf(tempFile, "%d,%s,%d\n", newuser.id, newuser.name, newuser.age);
+            }
+
+            fclose(file);
+            fclose(tempFile);
+
+            remove("user.txt");
+            rename("temp.txt", "user.txt");
+
+            if (flag)

```

Comment:
is it right name for variable 'flag'?
make changes at all places if applicable.